### DISCUSSION TITLE
Registering application to a custom URI Scheme

### DISCUSSION THREAD

**Discussion by Wiks00** [AvaloniaUI/Avalonia#9081] [Q&A]
Not too long ago I switched from MAUI to Avalonia and ran into a problem that I can't link `.appxmanifes` to a project to automatically register a custom url scheme in my app during local deployment (debugging).

Could anyone help me with this?

**Comment by maxkatz6** [comment]
> ran into a problem that I can't link .appxmanifes to a project to automatically register a custom url scheme in my app during local deployment (debugging).

Do you mean MSIX appxmanifest file? What exact problem did you find trying it?
There shouldn't be any problem with creating a ".wapproj" file (Windows Application Packaging Project type for MSIX packages) with configured manifest.

**Reply by maxkatz6** [reply]
If you seek solution for non-packages projects, Microsoft documentation and blogs should cover it 
https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/grant-identity-to-nonpackaged-apps
https://blogs.windows.com/windowsdeveloper/2019/10/29/identity-registration-and-activation-of-non-packaged-win32-apps/
We don't have a ready to use templates for that, as it's less related to the GUI framework.

**Reply by Wiks00** [reply]
Sorry for the confusion, I'm new to desktop development and don't know much about distributing Win applications.
 
I'm trying to figure out how to _add a custom URL scheme_ to my application for an oidc authentication callback. In a UWP app, this can be achieved by adding a custom protocol to the declaration section of the `.appxmanifes` file. Is there a similar solution for Avalonia?
